temporalio/client.py

```diff
@@ -5298,50 +5924,117 @@ async def start_workflow_update(
+                    update_workflow=update_req
+                ),
+            ],
+        )
+
+        # Repeatedly try to invoke ExecuteMultiOperation until the update
+        # reaches user-provided wait stage or is at least ACCEPTED (as of the
+        # time of this writing, the user cannot specify sooner than ACCEPTED)
+        seen_start = False
         while True:
             try:
-                resp = await self._client.workflow_service.update_workflow_execution(
-                    req,
-                    retry=True,
-                    metadata=input.rpc_metadata,
-                    timeout=input.rpc_timeout,
+                multiop_response = (
+                    await self._client.workflow_service.execute_multi_operation(
+                        multiop_req
+                    )
                 )
+            except asyncio.CancelledError as err:
+                raise WorkflowUpdateRPCTimeoutOrCancelledError() from err
             except RPCError as err:
-                if (
-                    err.status == RPCStatusCode.DEADLINE_EXCEEDED
-                    or err.status == RPCStatusCode.CANCELLED
-                ):
+                if err.status in [
+                    RPCStatusCode.DEADLINE_EXCEEDED,
+                    RPCStatusCode.CANCELLED,
+                ]:
                     raise WorkflowUpdateRPCTimeoutOrCancelledError() from err
                 else:
+                    multiop_failure = (
+                        temporalio.api.errordetails.v1.MultiOperationExecutionFailure()
+                    )
+                    if err.grpc_status.details[0].Unpack(multiop_failure):
+                        status = next(
+                            (
+                                st
+                                for st in multiop_failure.statuses
+                                if (
+                                    st.details
+                                    and not st.details[0].Is(
+                                        temporalio.api.failure.v1.MultiOperationExecutionAborted.DESCRIPTOR
+                                    )
+                                )
+                            ),
+                            None,
+                        )
+                        if status and status.code in RPCStatusCode:
+                            raise RPCError(
+                                status.message,
+                                RPCStatusCode(status.code),
+                                err.raw_grpc_status,
+                            )
                     raise
-            except asyncio.CancelledError as err:
-                raise WorkflowUpdateRPCTimeoutOrCancelledError() from err
+
+            start_responses: List[
+                temporalio.api.workflowservice.v1.StartWorkflowExecutionResponse
+            ] = [
+                r.start_workflow
+                for r in multiop_response.responses
```

Comment:
Doesn't really matter, but I think we should be able to assume the indexes of responses match 1:1 with the request objects. So may be able to just do `multiop_response.responses[0].start_workflow` and not have to loop.

Reply:
Yeah I decided that you're right and I didn't need to program so defensively here. So more minimal/cleaner now.